src/pkg/projects/Microsoft.NETCore.App.Ref/Microsoft.NETCore.App.Ref.pkgproj

```diff
@@ -39,4 +42,35 @@
     </ItemGroup>
   </Target>
 
+  <!--
+    Copy the files in the package's data/ dir to a layout directory. This is what the targeting pack
+    installer will place in the dotnet install dir.
+
+    This extracts from the nupkg (zip) directly. An alternative would be using the PackageFile
+    items, but there's some risk of handling TargetPath metadata differently than NuGet does. Using
+    the nupkg directly ensures results are identical.
+  -->
+  <Target Name="CreateTargetingPackLayout"
+          AfterTargets="CreatePackage">
+    <PropertyGroup>
+      <TargetingPackNupkgFile>$(PackageOutputPath)$(Id).$(PackageVersion).nupkg</TargetingPackNupkgFile>
+    </PropertyGroup>
+
+    <ZipFileGetEntries TargetArchive="$(TargetingPackNupkgFile)">
+      <Output TaskParameter="Entries" ItemName="TargetingPackNupkgEntries" />
+    </ZipFileGetEntries>
+
+    <ItemGroup>
+      <TargetingPackDataEntries
+        Include="@(TargetingPackNupkgEntries)"
+        Condition="$([System.String]::new('%(TargetingPackNupkgEntries.Identity)').StartsWith('data/'))" />
+    </ItemGroup>
+
+    <ZipFileExtractToDirectory
```

Comment:
Why do you pull them from the package?  Just a couple targets before this you have the entire contents in items (`@(File)` if I remember correctly).

Reply:
There's a bit of reasoning in the target comment:

> This extracts from the nupkg (zip) directly. An alternative would be using the PackageFile    items, but there's some risk of handling TargetPath metadata differently than NuGet does. Using the nupkg directly ensures results are identical.

It is slower to decompress it, I'm just coding a little defensively here.

Reply:
I don't find that defensive, it's just adding complexity.  We define TargetPath and consume it, you can rely on it, of if you can't that's a bug that we fix.

Another thing to consider is inserting a layout phase between collecting assets and pack.  That way the content of the pack can be defined by the layout instead of vice-versa.

Reply:
That's true. I did add a file in https://github.com/dotnet/core-setup/pull/5168 that has a non-folder TargetPath, which is the kind of thing I wanted to avoid handling--but it looks like that's actually the only one, and I don't need to do it that way. I'll get rid of this.

A layout-first approach sounds much better. I'll shoot for that.